haystack_experimental/components/converters/image_utils.py

```diff
@@ -0,0 +1,278 @@
+
+    :param image: PIL Image object
+    :param mime_type: The mime type to save the image in. Default is "image/jpeg".
+    :return: Base64 string of the image
+    """
+    # Convert image to RGB if it has an alpha channel and we are saving as JPEG
+    if (mime_type == "image/jpeg" or mime_type == "image/jpg") and (
+        image.mode in ("RGBA", "LA") or (image.mode == "P" and "transparency" in image.info)
+    ):
+        image = image.convert("RGB")
+
+    buffered = BytesIO()
+    form = MIME_TO_FORMAT.get(mime_type)
+    if form is None:
+        logger.warning(
+            "Could not determine format for mime type {mime_type}. Defaulting to JPEG.",
+            mime_type=mime_type,
+        )
+        form = "JPEG"
+    image.save(buffered, format=form)
+    return base64.b64encode(buffered.getvalue()).decode("utf-8")
+
+
+def downsize_image(
+    image: Union["Image", "ImageFile"],
+    size: Tuple[int, int],
+) -> "Image":
+    """
+    Resize the image to be smaller while maintaining the aspect ratio.
+
+    The latency of vision models can be improved by downsizing images ahead of time to be less than the
+    maximum size they are expected to be. You can specify the size as a tuple of (num_pixels, num_pixels) or
+    as a dictionary with tuples of (num_pixels, num_pixels) as values and aspect ratios as keys. The value of the
+    key being closest to the image aspect ratio is used.
+
+    :param image: PIL Image object
+    :param size: Tuple of (num_pixels, num_pixels)
+    :return: Resized PIL Image object
+    """
+    longer_dim = max(image.size)
+    shorter_dim = min(image.size)
+    aspect_ratio = shorter_dim / longer_dim
+
+    if isinstance(size, dict):
+        nearest_aspect_ratio_key = min(size, key=lambda aspect_ratio_key: abs(aspect_ratio - aspect_ratio_key))
+        size = size[nearest_aspect_ratio_key]
+
+    max_longer_dim = max(size)
+    max_shorter_dim = min(size)
+    aspect_ratio_max = max_shorter_dim / max_longer_dim
+
+    if aspect_ratio <= aspect_ratio_max:
+        # If the aspect ratio is less than max, we can resize based on the longer dim
+        new_longer_dim = max_longer_dim
+        new_shorter_dim = int(new_longer_dim * aspect_ratio)
+    else:
+        # If the aspect ratio is larger than max, we have to resize based on the shorter dim
+        new_shorter_dim = max_shorter_dim
+        new_longer_dim = int(new_shorter_dim / aspect_ratio)
+
+    landscape = image.width > image.height
+    if landscape:
+        new_width = new_longer_dim
+        new_height = new_shorter_dim
+    else:
+        new_width = new_shorter_dim
+        new_height = new_longer_dim
+
+    # Don't resize image if it's already within requirements
+    if image.width <= new_width and image.height <= new_height:
+        return image
+
+    resized_image = image.resize((new_width, new_height))
+    return resized_image
+
+
+def read_image_from_pdf(
```

Comment:
```suggestion
def read_images_from_pdf(
```

this would clarify the goal of this function

Reply:
I've updated the name to be `convert_pdf_to_images`